app/src/main/java/com/example/bottomnav/DataDecoder.java

```diff
@@ -0,0 +1,56 @@
+package com.example.bottomnav;
+
+public interface DataDecoder {
+    static DataDecoder getPrimativeDecoder(PayLoadDataType dataType) {
+        switch (dataType) {
+            case Integer:
+                return new IntegerDecoder();
+            case Float:
+                return new FloatDecoder();
+            case Double:
+                return new DoubleDecoder();
+            case Long:
+                return new LongDecoder();
+            case uint8_t:
+                return new IntegerDecoder();
+            case uint16_t:
+                return new IntegerDecoder();
+            default:
+                return null;
+        }
+    }
+
+    static String getPrimativeValue(PayLoadDataType dataType, DataDecoder decoder) {
```

Comment:
I'd recommend refactoring this into each decoder's `decodeToString` method or something.

In general, I'd say that if you're trying to define specialized behavior for objects on a class hierarchy, define it on the classes by overriding. In `getPrimitiveDecoder` you don't have this option, since you don't have the decoder yet, but here you have the decoder.